dockers/docker-dhcp-server/cli/config/plugins/dhcp_server.py

```diff
@@ -1,13 +1,371 @@
 import click
 import utilities_common.cli as clicommon
 
+import ipaddress
+
 
 @click.group(cls=clicommon.AbbreviationGroup, name="dhcp_server")
 def dhcp_server():
     """config DHCP Server information"""
     pass
 
 
+@dhcp_server.group(cls=clicommon.AliasedGroup)
+def ipv4():
+    """Show ipv4 related dhcp_server info"""
+    pass
+
+
+@dhcp_server.group(cls=clicommon.AliasedGroup)
+def ipv6():
+    """Show ipv6 related dhcp_server info"""
+    pass
+
+
+def sanity_check_add(mode, lease_time, infer_gw_nm, gateway, netmask, dhcp_interface):
+    if mode != "PORT":
+        raise Exception("Only mode PORT is supported")
+    if not lease_time.isdigit():
+        raise Exception("lease_time is required and must be nonnegative integer")
+    if not inter_gw_nm:
+        ipaddress.ip_address(gateway)
+        ipaddress.ip_address(netmask)
+
+
+def infer_param_for_add(dhcp_interface):
+    pass
+
+
+@ipv4.command()
+@click.argument("dhcp_interface", required=True)
+@click.option("--mode", required=True)
+@click.option("--lease_time", required=False, default="900")
+@click.option("--infer_gw_nm", required=False, default=False, is_flag=True)
+@click.option("--gateway", required=False)
+@click.option("--netmask", required=False)
+@clicommon.pass_db
+def add(db, mode, lease_time, infer_gw_nm, gateway, netmask, dhcp_interface):
+    sanity_check_add(mode, lease_time, infer_gw_nm, gateway, netmask, dhcp_interface):
+    if infer_gw_nm:
+        gateway, netmask = infer_param_for_add(dhcp_interface)
```

Comment:
Add check to make sure gateway and netmask are given if infer_gw_nm is not given